erts/emulator/hipe/hipe_x86_signal.c

```diff
@@ -159,16 +159,10 @@
 
 #if !(defined(__GLIBC__) || defined(__DARWIN__) || defined(__NetBSD__) || defined(__FreeBSD__) || defined(__sun__))
 /*
- * Unknown libc -- assume musl.  Note: musl deliberately does not provide a musl-specific
- * feature test macro, so we cannot check for it.
- *
- * sigaction is a weak alias for __sigaction, which is a wrapper for __libc_sigaction.
- * There are libc-internal calls to __libc_sigaction which install handlers, so we must
- * override __libc_sigaction rather than __sigaction.
+ * Unknown libc -- assume musl.  Also assume signal delivery on musl is safe.
```

Comment:
"Also assume signal delivery on musl is safe." makes no sense.  The whole point of this file is that any and all signal delivery is unsafe unless directed to the alternate signal stack.

The effect of the subsequent changes is to disable all sigaction() and __libc_sigaction() overrides, which means that only those signal handlers registered at the point hipe_x86_signal() is called get fixed to use SA_ONSTACK, subsequent signal handlers are not caught so will be delivered to the current %ESP/%RSP stack, which isn't safe.

So, I don't think this is safe, but as it only affects unknown libcs (including musl), I cannot really veto it.  It might be better however to `#error` in this case and tell affected users to build with `--disable-hipe`.

Reply:
I was thinking about an error saying to disable hipe too. That may be the safest.

Reply:
yes, please change this PR to print an error message saying to disable HiPE.

Reply:
ok

Reply:
Is there a similar place that causes compilation to error and telling the user to change config to disable something I can see how to do this?

Reply:
Grep for `#error` or `# error` under erts and you'll find many examples.

Reply:
I can do it if you don't want to figure out how to do it. The best thing would probably be to just automatically disable hipe during configure when the above condition happens.

Reply:
If you want to do that way, where it automatically disables during configure, then yea, I can close this.